maps/biter_battles_v2/difficulty_vote.lua

```diff
@@ -45,7 +45,16 @@ local function poll_difficulty(player)
 	
 	local frame = player.gui.center.add { type = "frame", caption = "Vote global difficulty:", name = "difficulty_poll", direction = "vertical" }
 	for key, _ in pairs(difficulties) do
-		local b = frame.add({type = "button", name = tostring(key), caption = difficulties[key].name .. " (" .. difficulties[key].str .. ")"})
+	
+		local amount_voters_of_diff = 0
+		local voter_text = "vote"
+		for _, playerVoting in pairs(game.connected_players) do
+			if key == global.difficulty_player_votes[playerVoting.name] then
+				amount_voters_of_diff = amount_voters_of_diff + 1
+			end
+		end
+		if amount_voters_of_diff > 1 then voter_text = "votes" end
+		local b = frame.add({type = "button", name = tostring(key), caption = difficulties[key].name .. " (" .. difficulties[key].str .. ")" .. " : " .. amount_voters_of_diff .. " " .. voter_text})
```

Comment:
This is highly inefficient, for 40 connected players we get 7*40=280 passes. Also, string concatenation `..` is slow
```suggestion
	local vote_amounts = {0,0,0,0,0,0,0}
	for k, v in pairs(global.difficulty_player_votes) do
		vote_amounts[v] = vote_amounts[v] + 1
	end
	
	for key, difficulty in pairs(difficulties) do
		local caption = table.concat({difficulty.name, " (", difficulty.str, ")", " : ", vote_amounts[key]})
		local b = frame.add{type = "button", name = tostring(key), caption = caption}
```

This way we get 40+7=47 passes (if everyone voted), and we use faster `table.concat`
Sidenote: for `key, value in pairs()` allows for direct value addressing, so we can use better looking `difficulty.name` instead of `difficulties[key].name`
Personally, I'd get rid of the "votes" part and leave just the number, but that's up to you.

I haven't tested it, but I'm pretty sure it should work